contracts/libraries/Safe130To141Migration.sol

```diff
@@ -49,9 +60,7 @@ contract Safe130To141Migration is SafeStorage {
      * @notice Migrate to Safe 1.4.1 Singleton (L1) at `SAFE_141_SINGLETON`
      * @dev This function should only be called via a delegatecall to perform the upgrade.
      */
-    function migrate() public {
-        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
```

Comment:
AFAIU, this can be removed because `isContract(singleton)` implies that this was called through a `DELEGATECALL` (since this contract doesn't set storage itself).

If that is the case, shouldn't `address public immutable MIGRATION_SINGLETON` be removed as it is no longer used?

Reply:
Also, on a general note, is it OK to have this kind of implicit "check" or should it be more explicit (at the cost of a small amount of additional gas)?

Reply:
> If that is the case, shouldn't address public immutable MIGRATION_SINGLETON be removed as it is no longer used?

Good catch, thanks. I will remove it

> Also, on a general note, is it OK to have this kind of check be "implicit" or should it be more explicit (at the cost of a small amount of additional gas)?

I think it's OK to have an implicit check if it's explicitly mentioned somewhere. I will try to improve the documentation.